src/org/zaproxy/zap/extension/pscanrules/CacheControlScanner.java

```diff
@@ -50,14 +50,24 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
 
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		boolean found=true;
+		String evidence = null;
+
 		if (msg.getRequestHeader().isSecure() && !msg.getRequestHeader().getURI().toString().toLowerCase().endsWith(".css") && !msg.getResponseHeader().isImage() && msg.getResponseBody().length() > 0) {
 			Vector<String> cacheControl = msg.getResponseHeader().getHeaders(HttpHeader.CACHE_CONTROL);
 			if (cacheControl != null) {
 				for (String cacheControlDirective : cacheControl) {
 					if (cacheControlDirective.toLowerCase().indexOf("no-store") < 0 || cacheControlDirective.toLowerCase().indexOf("no-cache") < 0 || cacheControlDirective.toLowerCase().indexOf("must-revalidate") < 0 || cacheControlDirective.toLowerCase().indexOf("private") < 0) {
-						this.raiseAlert(msg, id, cacheControlDirective);
+						found=false;
+						evidence=cacheControlDirective; //The non-compliant directive

```

Comment:
You're right, and I did think of that last night, because it did seem to me like I was adding too much code :disappointed:  But I couldn't quickly find a way to go from a vector to a unique list. since then I found `Set<Object> set = new HashSet<>(vector);` and will try something like that tonight.